src/ui/server/controllers/insurance/policy/broker-confirm-address/index.test.ts

```diff
@@ -60,6 +61,66 @@ describe('controllers/insurance/policy/broker-confirm-address', () => {
   });
 
   describe('get', () => {
+    describe(`when application.broker does not have ${POSTCODE}, ${BUILDING_NUMBER_OR_NAME} or ${FIELD_ID}`, () => {
+      beforeEach(() => {
+        res.locals.application = {
+          ...mockApplication,
+          broker: {
+            ...mockApplication.broker,
+            [POSTCODE]: '',
+            [BUILDING_NUMBER_OR_NAME]: '',
+            [FIELD_ID]: '',
+          },
+        };
+      });
+
+      it(`should redirect to ${BROKER_DETAILS_ROOT}`, () => {
+        get(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(`${INSURANCE_ROOT}/${referenceNumber}${BROKER_DETAILS_ROOT}`);
+      });
+    });
+
+    describe(`when application.broker has ${POSTCODE} and ${BUILDING_NUMBER_OR_NAME}, no ${FIELD_ID}`, () => {
+      beforeEach(() => {
+        res.locals.application = {
+          ...mockApplication,
+          broker: {
+            ...mockApplication.broker,
+            [POSTCODE]: 'Mock postcode',
+            [BUILDING_NUMBER_OR_NAME]: 'Mock building name/number',
+            [FIELD_ID]: '',
+          },
+        };
+      });
+
+      it(`should NOT redirect to ${BROKER_DETAILS_ROOT}`, () => {
+        get(req, res);
+
+        expect(res.redirect).toHaveBeenCalledTimes(0);
+      });
+    });
+
+    describe(`when application.broker has ${FIELD_ID}, no ${POSTCODE} or ${BUILDING_NUMBER_OR_NAME}`, () => {
+      beforeEach(() => {
```

Comment:
Can we add an assertion when there is no `BUILDING_NUMBER_OR_NAME` but has `FIELD_ID` and `POSTCODE` please.

Reply:
Good shout, have added tests for when there is:
- `FIELD_ID` and `POSTCODE , no`BUILDING_NUMBER_OR_NAME`
- `FIELD_ID` and `BUILDING_NUMBER_OR_NAME , no`POSTCODE`